### PR TITLE
Add TraceIdLoggingFilter

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Comment by uschi2000** [comment]
@schlosna : thoughts on closing the MDC put? (you had made the corresponding change earlier, if I remember right)

**Comment by uschi2000** [comment]
As far as I know, this is the last thing that needs to go in before we can make another RC.

File: jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/HttpRemotingJerseyFeature.java [hunk 1]
```diff
@@ -51,6 +51,9 @@ public boolean configure(FeatureContext context) {
 
         // Tracing
         context.register(new TraceEnrichingFilter());
+//        environment.servlets()
+//                -                .addFilter(TraceIdLoggingFilter.class.getSimpleName(), TraceIdLoggingFilter.INSTANCE)
+//                -                .addMappingForUrlPatterns(EnumSet.allOf(DispatcherType.class), true, "/*");
```
Comment (markelliot) [inline jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/HttpRemotingJerseyFeature.java]: just delete?

File: jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java [hunk 2]
```diff
@@ -53,6 +59,10 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                 Tracer.startSpan(operation, spanId); // caller's span is this span's parent.
             }
         }
+
+        // Give SLF4J appenders access to the trace id
+        // TODO(rfink) We should use putCloseable; when and how can we remove it though? There is no filter chain.
+        MDC.put(MDC_KEY, Tracer.getTraceId());
```
Comment (markelliot) [inline jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java:65]: in addition to this we should add to the request attributes, e.g.:

```
// convention is package naming
requestContext.setProperty("com.palantir.remoting1.traceId", Tracer.getTraceId());
```

Comment (markelliot) [reply jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java:65]: (this allows picking things up in the Access logger, assuming it's not asynchronous or logback fixes the bug I submitted a patch for)

File: jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/HttpRemotingJerseyFeature.java (hunk 1 above)
Comment (uschi2000) [reply jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/HttpRemotingJerseyFeature.java]: done

File: jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java (hunk 2 above)
Comment (uschi2000) [reply jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java:65]: assuming it's not _synchronous_?

Comment (uschi2000) [reply jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java:65]: done

Comment (markelliot) [reply jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java:65]: actually meant synchronous, but whether they're sync or async shouldn't actually matter (modulo fact a bug exists)

**Review by markelliot — APPROVED** [review]

**Pull request merged**